### PR TITLE
Issue 510: Frontend local environment fixes/documentation

### PR DESCRIPTION
**Pull request by jtwillis92**
### This pull request changes...

Fixes issues with loading in the appropriate environment variables by adding in the default `.env` file and documenting the method by which the frontend applies `.env` files.

## FUNCTIONALITY
Addresses the frontend side of #510 

## TO TEST

*List the steps to test the PR*

- Delete any `.env.local` files you might have for the frontend
- Unset any environment variables on your local machine
- Run `docker-compose up -d` in the frontend directory and confirm the frontend works as expected (try to login, navigate between pages, etc)

The frontend should just work with the default `.env` file and then any local overrides can be applied in a `.env.local` file (Do not commit local overrides!)

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x] `dotenv` files are removed from the repositories with `.gitignore` (NOTE: only .env.local overrides are ignored for the frontend only)
  - [x] `dotenv` files are used locally for storing environment variables
  - [x] Deployment strategy is updated to use another solution for handling environment variables
  - [x] Testing Checklist has been run and all tests pass
  
As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [x] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?
+ [x] If this PR introduces frontend code, is that code documented both inline and overall?
+ [x] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### CODE REVIEW

File: tdrs-frontend/.env.development
Comment (jtwillis92) [inline tdrs-frontend/.env.development:3]: Note, none of these values are different than the defaults. We could leave this file empty and it would just defer to the default `.env` file. 

Also, this file only gets loaded when running npm or yarn start directly (ie. no docker) - not to be confused with the `development` environment in cloud.gov

File: tdrs-frontend/.env.test
Comment (jtwillis92) [inline tdrs-frontend/.env.test:3]: Same here, a matter of preference but we could leave this file blank as well

File: tdrs-frontend/README.md
Comment (jtwillis92) [inline tdrs-frontend/README.md:53]: This may be important to bring up with 18F, due to the way the dockerfile deploys the app React acts as if its in a `production` environment. This doesn't technically affect anything at the moment, but if we were to commit a `.env.production` file with variables specific to production the `.env.local` file would need to be present for all local devs in order to prevent environment issues.

Comment (jtwillis92) [reply tdrs-frontend/README.md:53]: @jorgegonzalez opened https://github.com/raft-tech/TANF-app/issues/691 to address this issue

File: tdrs-frontend/README.md
Comment (jorgegonzalez) [inline tdrs-frontend/README.md:67]:
```suggestion
#### `npm test` / `yarn test`
```

nit: add the `yarn` equivalent for consistency.

**Review by jorgegonzalez — APPROVED** [review]
LGTM

**Review by carltonsmith — APPROVED** [review]
LGTM

**Comment by alexsoble** [comment]
I'll be reviewing this one while @ADPennington gets her local env set up!

File: .gitignore
Comment (alexsoble) [inline .gitignore:15]: 👍 👍

File: tdrs-frontend/.env
Comment (alexsoble) [inline tdrs-frontend/.env:3]: Love the link here, all the context a dev needs to understand what's going on! 😍

File: tdrs-frontend/.env
Comment (alexsoble) [inline tdrs-frontend/.env:4]: Good warning

File: tdrs-frontend/.env.test
Comment (alexsoble) [reply tdrs-frontend/.env.test:3]: Or even delete it? (But — these are potential future simplifications, let's not let them hold up this PR!)

File: tdrs-frontend/README.md
Comment (alexsoble) [reply tdrs-frontend/README.md:53]: Thank you for spotting this and looking ahead to it!

**Comment by alexsoble** [comment]
# QASP Review

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Tested frontend ENV variables locally — it worked! 😄 
+ [x] Tested local override with an `.env.local` — it worked! 😄  

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

_N/A, infrastructure change._

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

_N/A, infrastructure change._

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

 _N/A, infrastructure change._

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] Excellent documentation additions to frontend README
+ [x] Inline documentation added to frontend .env files

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] OWASP Scan pass

**Review by alexsoble — APPROVED** [review]

**Comment by ADPennington** [comment]
pairsed w/ @alexsoble for this QASP review 👍🏾

**Pull request merged**
